midi/midi/decode.py

```diff
@@ -18,25 +20,26 @@ class EventNote:
 
     def __init__(
             self,
-            velocity: int | float,
+            velocity: Union[int, float],
             height: int
     ):
         self.velocity = float(velocity)
         self.height = height
         self.tone = height % 12 + 1
         self.octave = height // 12 - 1
 
-    def __eq__(self, other):
+    def __eq__(self,
+               other: Any) -> bool:
         return self.velocity == other.velocity and \
                self.height == other.height and \
                self.tone == other.tone and \
                self.octave == other.octave
 
-    def __repr__(self):
+    def __repr__(self) -> str:
         return f'EventNote({self.velocity}, {self.height})'
 
     def normalise(self,
-                  max_velocity: int | float):
+                  max_velocity: Union[int, float]) -> None:
```

Comment:
etc

Reply:
it turns out a component (don't remember which one) fails at complex structures to differentiate logical "|" from mathematical "|" (CI threw a few errors)